From 03451c7f5be641d89c63dec8ee07ab4f9e275ebb Mon Sep 17 00:00:00 2001
From: Ryan O'Hara <rohara@redhat.com>
Date: Tue, 16 Dec 2014 13:25:55 -0600
Subject: [PATCH] Fix format of long int in log_message call

---
 keepalived/vrrp/vrrp_snmp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/keepalived/vrrp/vrrp_snmp.c b/keepalived/vrrp/vrrp_snmp.c
index 6df7795..ef8ef97 100644
--- a/keepalived/vrrp/vrrp_snmp.c
+++ b/keepalived/vrrp/vrrp_snmp.c
@@ -541,7 +541,7 @@ vrrp_snmp_instance_priority(int action,
 		/* Commit: change values. There is no way to fail. */
 		log_message(LOG_INFO,
 			    "VRRP_Instance(%s) base priority changed from"
-			    " %d to %d via SNMP.",
+			    " %d to %ld via SNMP.",
 			    vrrp->iname, vrrp->base_priority, (long)(*var_val));
 		vrrp->base_priority = (long)(*var_val);
 		/* If we the instance is not part of a sync group, the
-- 
1.9.3

