From 8228f6eea861ca293a868dee06c564756009785c Mon Sep 17 00:00:00 2001
From: Quentin Armitage <quentin@armitage.org.uk>
Date: Tue, 15 Aug 2017 20:38:42 +0100
Subject: [PATCH] Fix sense of a conditional compilation check

Signed-off-by: Quentin Armitage <quentin@armitage.org.uk>
---
 keepalived/vrrp/vrrp_scheduler.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/keepalived/vrrp/vrrp_scheduler.c b/keepalived/vrrp/vrrp_scheduler.c
index 48acd11..9d33a62 100644
--- a/keepalived/vrrp/vrrp_scheduler.c
+++ b/keepalived/vrrp/vrrp_scheduler.c
@@ -661,7 +661,7 @@ vrrp_backup(vrrp_t * vrrp, char *buffer, ssize_t len)
 
 static void
 vrrp_become_master(vrrp_t * vrrp,
-#ifdef _WITH_VRRP_AUTH_
+#ifndef _WITH_VRRP_AUTH_
 				 __attribute__((unused))
 #endif
 							 char *buffer, __attribute__((unused)) ssize_t len)
-- 
2.9.5

