From 3b83f50634061216eda32a1dbc5601f1b69d7d60 Mon Sep 17 00:00:00 2001
From: Quentin Armitage <quentin@armitage.org.uk>
Date: Fri, 1 Sep 2017 14:14:25 +0100
Subject: [PATCH] Generate README from README.md

Signed-off-by: Quentin Armitage <quentin@armitage.org.uk>
---
 .gitignore  |   1 +
 Makefile.am |  13 ++++++++
 Makefile.in | 102 ++++++++++++++++++++++++++++++++++++++++++++++++++----------
 3 files changed, 100 insertions(+), 16 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index fecba88..c706ffb 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -8,6 +8,9 @@ export			DEBUG_CFLAGS
 export			DEBUG_CPPFLAGS
 export			DEBUG_LDFLAGS
 
+edit = echo "  EDIT     $@"; \
+	 sed -e "/^\[\!\[/d"
+
 SUBDIRS 		= lib keepalived doc
 
 if BUILD_GENHASH
@@ -18,12 +21,22 @@ SUBDIRS			+= bin_install
 
 EXTRA_DIST		= AUTHOR CONTRIBUTORS snap README.md
 
+doc_DATA		= README
+
+MOSTLYCLEANFILES	= README
+
 MAINTAINERCLEANFILES	= @MAINTAINERCLEANFILES@
 
+README:	$(srcdir)/README.md
+	@$(edit) '$(srcdir)/$@.md' >$@
+
 distclean-local:
 	@rm -f aclocal.m4 ar-lib compile depcomp missing keepalived-$(VERSION).tar.gz config.log config.status
 	@rm -rf autom4te.cache
 
+dist-hook:
+	@rm -f $(distdir)/README
+
 .PHONY: docker
 docker:
 	docker build -t keepalived .
diff --git a/Makefile.in b/Makefile.in
index 896845a..9daefec 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -19,6 +19,7 @@
 # Keepalived OpenSource project.
 #
 # Copyright (C) 2001-2016 Alexandre Cassen, <acassen@gmail.com>
+
 VPATH = @srcdir@
 am__is_gnu_make = { \
   if test -z '$(MAKELEVEL)'; then \
@@ -132,6 +133,35 @@ am__can_run_installinfo = \
     n|no|NO) false;; \
     *) (install-info --version) >/dev/null 2>&1;; \
   esac
+am__vpath_adj_setup = srcdirstrip=`echo "$(srcdir)" | sed 's|.|.|g'`;
+am__vpath_adj = case $$p in \
+    $(srcdir)/*) f=`echo "$$p" | sed "s|^$$srcdirstrip/||"`;; \
+    *) f=$$p;; \
+  esac;
+am__strip_dir = f=`echo $$p | sed -e 's|^.*/||'`;
+am__install_max = 40
+am__nobase_strip_setup = \
+  srcdirstrip=`echo "$(srcdir)" | sed 's/[].[^$$\\*|]/\\\\&/g'`
+am__nobase_strip = \
+  for p in $$list; do echo "$$p"; done | sed -e "s|$$srcdirstrip/||"
+am__nobase_list = $(am__nobase_strip_setup); \
+  for p in $$list; do echo "$$p $$p"; done | \
+  sed "s| $$srcdirstrip/| |;"' / .*\//!s/ .*/ ./; s,\( .*\)/[^/]*$$,\1,' | \
+  $(AWK) 'BEGIN { files["."] = "" } { files[$$2] = files[$$2] " " $$1; \
+    if (++n[$$2] == $(am__install_max)) \
+      { print $$2, files[$$2]; n[$$2] = 0; files[$$2] = "" } } \
+    END { for (dir in files) print dir, files[dir] }'
+am__base_list = \
+  sed '$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;s/\n/ /g' | \
+  sed '$$!N;$$!N;$$!N;$$!N;s/\n/ /g'
+am__uninstall_files_from_dir = { \
+  test -z "$$files" \
+    || { test ! -d "$$dir" && test ! -f "$$dir" && test ! -r "$$dir"; } \
+    || { echo " ( cd '$$dir' && rm -f" $$files ")"; \
+         $(am__cd) "$$dir" && rm -f $$files; }; \
+  }
+am__installdirs = "$(DESTDIR)$(docdir)"
+DATA = $(doc_DATA)
 RECURSIVE_CLEAN_TARGETS = mostlyclean-recursive clean-recursive	\
   distclean-recursive maintainer-clean-recursive
 am__recursive_targets = \
@@ -162,7 +192,7 @@ CTAGS = ctags
 CSCOPE = cscope
 DIST_SUBDIRS = lib keepalived doc genhash bin_install
 am__DIST_COMMON = $(srcdir)/Makefile.in $(srcdir)/keepalived.spec.in \
-	COPYING ChangeLog INSTALL TODO ar-lib compile depcomp \
+	COPYING ChangeLog INSTALL README TODO ar-lib compile depcomp \
 	install-sh missing
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 distdir = $(PACKAGE)-$(VERSION)
@@ -314,8 +344,13 @@ target_alias = @target_alias@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
+edit = echo "  EDIT     $@"; \
+	 sed -e "/^\[\!\[/d"
+
 SUBDIRS = lib keepalived doc $(am__append_1) bin_install
 EXTRA_DIST = AUTHOR CONTRIBUTORS snap README.md
+doc_DATA = README
+MOSTLYCLEANFILES = README
 DISTCHECK_CONFIGURE_FLAGS = \
         --with-systemdsystemunitdir=$$dc_install_base/$(systemdsystemunitdir)
 
@@ -357,6 +392,27 @@ $(ACLOCAL_M4):  $(am__aclocal_m4_deps)
 $(am__aclocal_m4_deps):
 keepalived.spec: $(top_builddir)/config.status $(srcdir)/keepalived.spec.in
 	cd $(top_builddir) && $(SHELL) ./config.status $@
+install-docDATA: $(doc_DATA)
+	@$(NORMAL_INSTALL)
+	@list='$(doc_DATA)'; test -n "$(docdir)" || list=; \
+	if test -n "$$list"; then \
+	  echo " $(MKDIR_P) '$(DESTDIR)$(docdir)'"; \
+	  $(MKDIR_P) "$(DESTDIR)$(docdir)" || exit 1; \
+	fi; \
+	for p in $$list; do \
+	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
+	  echo "$$d$$p"; \
+	done | $(am__base_list) | \
+	while read files; do \
+	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(docdir)'"; \
+	  $(INSTALL_DATA) $$files "$(DESTDIR)$(docdir)" || exit $$?; \
+	done
+
+uninstall-docDATA:
+	@$(NORMAL_UNINSTALL)
+	@list='$(doc_DATA)'; test -n "$(docdir)" || list=; \
+	files=`for p in $$list; do echo $$p; done | sed -e 's|^.*/||'`; \
+	dir='$(DESTDIR)$(docdir)'; $(am__uninstall_files_from_dir)
 
 # This directory's subdirectories are mostly independent; you can cd
 # into them and run 'make' without going through this Makefile.
@@ -521,6 +577,9 @@ distdir: $(DISTFILES)
 	      || exit 1; \
 	  fi; \
 	done
+	$(MAKE) $(AM_MAKEFLAGS) \
+	  top_distdir="$(top_distdir)" distdir="$(distdir)" \
+	  dist-hook
 	-test -n "$(am__skip_mode_fix)" \
 	|| find "$(distdir)" -type d ! -perm -755 \
 		-exec chmod u+rwx,go+rx {} \; -o \
@@ -654,9 +713,12 @@ distcleancheck: distclean
 	       exit 1; } >&2
 check-am: all-am
 check: check-recursive
-all-am: Makefile
+all-am: Makefile $(DATA)
 installdirs: installdirs-recursive
 installdirs-am:
+	for dir in "$(DESTDIR)$(docdir)"; do \
+	  test -z "$$dir" || $(MKDIR_P) "$$dir"; \
+	done
 install: install-recursive
 install-exec: install-exec-recursive
 install-data: install-data-recursive
@@ -677,6 +739,7 @@ install-strip:
 	    "INSTALL_PROGRAM_ENV=STRIPPROG='$(STRIP)'" install; \
 	fi
 mostlyclean-generic:
+	-test -z "$(MOSTLYCLEANFILES)" || rm -f $(MOSTLYCLEANFILES)
 
 clean-generic:
 
@@ -710,7 +773,7 @@ info: info-recursive
 
 info-am:
 
-install-data-am:
+install-data-am: install-docDATA
 
 install-dvi: install-dvi-recursive
 
@@ -756,25 +819,26 @@ ps: ps-recursive
 
 ps-am:
 
-uninstall-am:
+uninstall-am: uninstall-docDATA
 
 .MAKE: $(am__recursive_targets) install-am install-strip
 
 .PHONY: $(am__recursive_targets) CTAGS GTAGS TAGS all all-am \
 	am--refresh check check-am clean clean-cscope clean-generic \
 	clean-local cscope cscopelist-am ctags ctags-am dist dist-all \
-	dist-bzip2 dist-gzip dist-lzip dist-shar dist-tarZ dist-xz \
-	dist-zip distcheck distclean distclean-generic distclean-local \
-	distclean-tags distcleancheck distdir distuninstallcheck dvi \
-	dvi-am html html-am info info-am install install-am \
-	install-data install-data-am install-dvi install-dvi-am \
-	install-exec install-exec-am install-html install-html-am \
-	install-info install-info-am install-man install-pdf \
-	install-pdf-am install-ps install-ps-am install-strip \
-	installcheck installcheck-am installdirs installdirs-am \
-	maintainer-clean maintainer-clean-generic mostlyclean \
-	mostlyclean-generic pdf pdf-am ps ps-am tags tags-am uninstall \
-	uninstall-am
+	dist-bzip2 dist-gzip dist-hook dist-lzip dist-shar dist-tarZ \
+	dist-xz dist-zip distcheck distclean distclean-generic \
+	distclean-local distclean-tags distcleancheck distdir \
+	distuninstallcheck dvi dvi-am html html-am info info-am \
+	install install-am install-data install-data-am \
+	install-docDATA install-dvi install-dvi-am install-exec \
+	install-exec-am install-html install-html-am install-info \
+	install-info-am install-man install-pdf install-pdf-am \
+	install-ps install-ps-am install-strip installcheck \
+	installcheck-am installdirs installdirs-am maintainer-clean \
+	maintainer-clean-generic mostlyclean mostlyclean-generic pdf \
+	pdf-am ps ps-am tags tags-am uninstall uninstall-am \
+	uninstall-docDATA
 
 .PRECIOUS: Makefile
 
@@ -783,10 +847,16 @@ export			DEBUG_CFLAGS
 export			DEBUG_CPPFLAGS
 export			DEBUG_LDFLAGS
 
+README:	$(srcdir)/README.md
+	@$(edit) '$(srcdir)/$@.md' >$@
+
 distclean-local:
 	@rm -f aclocal.m4 ar-lib compile depcomp missing keepalived-$(VERSION).tar.gz config.log config.status
 	@rm -rf autom4te.cache
 
+dist-hook:
+	@rm -f $(distdir)/README
+
 .PHONY: docker
 docker:
 	docker build -t keepalived .
-- 
2.9.5

