diff -Naupr keepalived-1.1.13.orig/keepalived/vrrp/vrrp_arp.c keepalived-1.1.13/keepalived/vrrp/vrrp_arp.c
--- keepalived-1.1.13.orig/keepalived/vrrp/vrrp_arp.c	2006-10-11 11:44:59.000000000 +0200
+++ keepalived-1.1.13/keepalived/vrrp/vrrp_arp.c	2007-03-22 16:22:29.000000000 +0100
@@ -22,14 +22,14 @@
  * Copyright (C) 2001-2006 Alexandre Cassen, <acassen@linux-vs.org>
  */
 
-/* system includes */
-#include <linux/if_packet.h>
-
 /* local includes */
 #include "vrrp_arp.h"
 #include "memory.h"
 #include "utils.h"
 
+/* system includes */
+#include <linux/if_packet.h>
+
 /* global vars */
 char *garp_buffer;
 int garp_fd;
